src/main/java/sqlitedb/SQLiteDB.java

```diff
@@ -3,25 +3,40 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.sqlite.SQLiteConnection;
+import sandbox.ArchivableFile;
 import sandbox.SqlSandboxUtils;
 import services.ConnectionHandler;
 
 import javax.sql.DataSource;
-import java.io.File;
+import java.io.IOException;
 import java.sql.Connection;
 import java.sql.SQLException;
 
 public class SQLiteDB implements ConnectionHandler {
     private DBPath dbPath;
+    private ArchivableFile dbArchivableFile;
     private final Log log = LogFactory.getLog(SQLiteDB.class);
     private Connection connection;
 
     public SQLiteDB(DBPath dbPath) {
         this.dbPath = dbPath;
+        /*
+           FormplayerStorageFactory and RestoreFactory are instantiated with sqLiteDB = SQLLiteDB(null)
+           and sqLiteDB is only set to a real value during .configure;
+           this doesn't make a lot of sense to me, but appears to be required for tests.
+        */
+        if (dbPath != null) {
+            dbArchivableFile = new ArchivableFile(dbPath.getDatabaseFile());
+        }
     }
 
     private Connection getNewConnection() throws SQLException {
-        DataSource dataSource = SqlSandboxUtils.getDataSource(dbPath.getDatabaseName(), dbPath.getDatabasePath());
+        try {
+            dbArchivableFile.unarchiveIfArchived();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
The idea is that we only reach this line if we had an archive, but not a DB, and then failed to expand that DB, right? If so, then I'm thinking it might be better to just delete the folder here and re-instantiate a new database connection rather than failing hard